nodeutil/reflect.go

```diff
@@ -515,6 +515,13 @@ func (self Reflect) WriteFieldWithFieldName(fieldName string, m meta.Leafable, p
 	}
 
 	fieldVal := elemVal.FieldByName(fieldName)
+	if !fieldVal.IsValid() {
+		panic(fmt.Sprintf("Invalid property \"%s\" on %s", fieldName, elemVal.Type()))
+	}
+	if v == nil {
+		panic(fmt.Sprintf("No value given to set %s", m.Ident()))
+	}
+
```

Comment:
Not sure there is much you can do if reflect is not working. The reason I moved the check was that a lot of my early experiments with When() functions threw a panic because many reflect functions don't work on nil or "invalid" reflect values.

Reply:
correct.  When reflection is  invalid and When.OnRead and When.OnWrite is still called then implementations would call/read methods/field directly w/o using reflection.  I do see why you thought this was a bug tho.  If we change it now, it would change valid (admit-tingly not intuitive) behavior.

Reply:
then we should at least add some warning comments that the input can be invalid/nil... so everyone knows that the functions have to check.

Reply:
agreed.